docs/openapi.yml

```diff
@@ -0,0 +1,1367 @@
+            pattern: ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
+      responses:
+        '200':
+          description: A paginated list of attacks received by the host.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/Pagination'
+                  - type: object
+                    properties:
+                      data:
+                        type: array
+                        items:
+                          type: object
+                          additionalProperties: false
+                          properties:
+                            id:
+                              type: string
+                              example: "ab733030-e20d-4c58-9520-76debc1ad067"
+                              pattern: ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
+                            type:
+                              type: string
+                              example: "fail2ban"
+                              pattern: ^fail2ban$|^portsentry$
+                            node_id:
+                              type: string
+                              example: "f0f91a2d-81bd-491a-8531-9e36d332cbf0"
+                              pattern: ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
+                            ip:
+                              type: string
+                              format: ipv4
+                              example: "80.74.35.22/32"
+                              pattern: ^([0-9]{1,3}\.){3}[0-9]{1,3}(\/([0-9]|[1-2][0-9]|3[0-2]))?$
+                            status:
+                              type: boolean
+                              example: false
+                            first:
+                              type: string
+                              format: date-time
+                              example: "2017-02-11T14:55:08Z"
+                              maxLength: 20
+                            last:
+                              type: string
+                              format: date-time
+                              example: "2017-02-15T14:22:08Z"
+                              maxLength: 20
+                            count:
+                              type: integer
+                              example: 34
+                        maxItems: 15
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '403':
+          $ref: '#/components/responses/ForbiddenError'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+        '406':
+          $ref: '#/components/responses/NotAcceptableError'
+        '415':
+          $ref: '#/components/responses/UnsupportedMediaType'
+        '429':
+          $ref: '#/components/responses/TooManyRequests'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+  /attacks/ip/{ip}:
```

Comment:
nit: can't we have some query parameters that let us narrow the scope? We would reduce the number of our endpoints
Eg having an optional hostid parameter and ip parameter, and let the backend handle that